packages/graph/hash_graph/lib/graph/src/api/rest/entity_type.rs

```diff
@@ -130,23 +134,62 @@ async fn create_entity_type<P: StorePool + Send>(
         .map(Json)
 }
 
+#[derive(Deserialize, Component)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+struct EntityTypeQuery {
+    query: Expression,
+    data_type_query_depth: u8,
+    property_type_query_depth: u8,
+    link_type_query_depth: u8,
+    entity_type_query_depth: u8,
+}
+
 #[utoipa::path(
     post,
     path = "/entity-types/query",
-    request_body = Expression,
+    request_body = EntityTypeQuery,
     tag = "EntityType",
     responses(
-        (status = 200, content_type = "application/json", description = "List of all entity types matching the provided query", body = [PersistedEntityType]),
+        (status = 200, content_type = "application/json", description = "List of all entity types matching the provided query", body = [EntityTypeTree]),
```

Comment:
Do we want to rephrase this description a little perhaps?

Reply:
Improved in 9ff07d30ddaaad3344bdf8c8a59b1d346c41146a